### PR TITLE
feat(query): support vacuum temporary files

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

feat(query): support vacuum temporary files
```
Syntax: 
VACUUM TEMPORARY FILES [RETAIN \<number\> SECONDS|DAYS] [LIMIT \<number\>]; 
    RETAIN: temporary files exceeding this time range will be deleted.
    LIMIT: the maximum number of files to be deleted in this operation. helps for situations where there are too many temporary files.
```

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14690)

### CODE REVIEW

File: src/query/ee/src/storages/fuse/operations/vacuum_temporary_files.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    let mut remove_temp_files_name = Vec::new();
+    while let Some(de) = ds.try_next().await? {
+        let meta = de.metadata();
+
+        if let Some(modified) = meta.last_modified() {
+            if timestamp - modified.timestamp() >= expire_time {
+                operator.delete(de.path()).await?;
+                remove_temp_files_name.push(de.name().to_string());
+            }
```
Comment (sundy-li) [inline src/query/ee/src/storages/fuse/operations/vacuum_temporary_files.rs:109]: break if `timestamp` is new enough?

Comment (zhang2014) [reply src/query/ee/src/storages/fuse/operations/vacuum_temporary_files.rs:58]: operator listers are not ordered.

**Comment by zhang2014** [comment]
> Better to introcude the full sytax of the new feature, it's helpful to write document.

Done. CC: @soyeric128

**Review by sundy-li — APPROVED** [review]

**Review by lichuang — COMMENTED** [review]
it is better to add some unit test cases like in `src/query/ee/tests/it/storages/fuse/operations/vacuum.rs`

**Review by zhyass — APPROVED** [review]

**Comment by zhang2014** [comment]
> it is better to add some unit test cases like in `src/query/ee/tests/it/storages/fuse/operations/vacuum.rs`

Done.

**Pull request merged**